src/library/scala/math/BigInt.scala

```diff
@@ -14,16 +14,30 @@ package scala
 package math
 
 import java.math.BigInteger
-
-import scala.annotation.nowarn
+import scala.annotation.{nowarn, unused}
 import scala.language.implicitConversions
 import scala.collection.immutable.NumericRange
 
 object BigInt {
 
+  private final val bigTag = java.lang.Long.MIN_VALUE
```

Comment:
why do you use MIN_VALUE here, but Long.MinValue below? Is it because you are leveraging inlining of `final val` without type? If so, maybe comment so no one changes it later. Even so, is it somehow better to use that vs `Long.MinValue`?

Reply:
I tried to make the tag scheme explicit, but it actually muddies the waters. Reverted